src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/BotSegmentsPlaybackController.cs

```diff
@@ -359,11 +366,20 @@ private void ProcessBotSegmentAction(BotSegment firstActionSegment, Dictionary<l
                     _explorationDriver.StopExploring(firstActionSegment.Replay_SegmentNumber);
                     _explorationDriver.ReportPreviouslyCompletedAction(firstActionSegment.botAction.data);
                 }
-
+                
+                // Run (and potentially end) the validations for this segment
+                firstActionSegment.ProcessValidation();
```

Comment:
ProcessValidations calls should be moved back to ProcessBotSegments, just after the action, but not embedded in ProcessBotSegmentAction